Framework/PythonInterface/mantid/plots/__init__.py

```diff
@@ -763,6 +775,48 @@ def tricontourf(self, *args, **kwargs):
         else:
             return Axes.tricontourf(self, *args, **kwargs)
 
+    def legend(self, *args, **kwargs):
+        """
+        In the case where there are no arguments to the legend function,
+        but errors are present on the plot, we make our own legend order
+        to prevent MPL always appending errors on the bottom of the legend.
```

Comment:
Could you help me understand the scenario where MPL appends the errors on the bottom of the legend? I haven't seen that before.

Reply:
It seems that is the possibly unintended, but default behaviour when you have a mixture of lines that do not have `ErrorbarContainer`s associated with them, and lines that do. It happens because the `ax.containers` are added last [here](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/legend.py#L1212), which makes lines with errors always appear below lines without errors.